cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/InMemoryServiceProgramRunner.java

```diff
@@ -58,8 +60,8 @@ public ProgramController run(Program program, ProgramOptions options) {
     ServiceSpecification serviceSpec = appSpec.getServices().get(program.getName());
     Preconditions.checkNotNull(serviceSpec, "Missing ServiceSpecification for %s", program.getName());
 
-    //RunId for the service
-    RunId runId = RunIds.generate();
+    // RunId for the service
+    RunId runId = ProgramRunners.getRunId(options);
```

Comment:
This was a harmless bug that went unnoticed until I added the starting state. For programs, we created a RunId already in `AbstractProgramRuntimeService`. But `InMemoryServiceProgramRunner`, we recreated it for some reason, but it didn't matter since the service wasn't running either way.

In other words, it was always available via the program options.